packages/database/src/migrations/20210128055252-AddStrivePositiveResultBarGraph-modifies-data.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+import { insertObject } from '../utilities/migration';
+import { generateId } from '../utilities/generateId';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+
+const CODE_TO_NAME = { STR_PF05: 'PF', STR_PV05: 'PV', STR_PM05: 'PM', STR_PO05: 'PO' };
+
+const insertIndicator = (db, code) => {
+  const indicator = {
+    id: generateId(),
+    code: `STRIVE_Percentage_${CODE_TO_NAME[code]}_Of_Total_Weekly`,
+    builder: 'arithmetic',
+    config: {
+      formula: `${code} / STR_QMAL05`,
+      aggregation: 'SUM_EACH_WEEK',
+    },
+  };
+  insertObject(db, 'indicator', indicator);
+};
+
+const REPORT = {
+  id: 'PG_Strive_PNG_Positive_Result_By_Type_Bar',
+  dataBuilder: 'analyticsPerPeriod',
+  dataBuilderConfig: {
+    series: [
+      { key: 'Pf +ve', dataElementCode: 'STRIVE_Percentage_PF_Of_Total_Weekly' },
+      { key: 'Pv +ve', dataElementCode: 'STRIVE_Percentage_PV_Of_Total_Weekly' },
+      { key: 'Pm +ve', dataElementCode: 'STRIVE_Percentage_PM_Of_Total_Weekly' },
+      { key: 'Po +ve', dataElementCode: 'STRIVE_Percentage_PO_Of_Total_Weekly' },
+    ],
+    entityAggregation: {
+      dataSourceEntityType: 'village',
+      aggregationEntityType: 'facility',
+      aggregationType: 'REPLACE_ORG_UNIT_WITH_ORG_GROUP',
+    },
+    programCodes: ['SQMAL', 'SPF', 'SPV', 'SPM', 'SPO'],
```

Comment:
Data Broker does something funny (in pullAnalyticsFromEventsForApi). It lets you call getAnalytics with multiple program codes. 

BEFORE: It used to be that it would just loop the program codes and for each program request all the DE codes.  

AFTER: After these changes, it loops the program codes and for each program it only requests the DE codes for that program.

-----

Even though pullAnalyticsFromEventsForApi lets you do this, no vis/data builder does this, or it would fall over. TODO: check if this anyone actually does this (look at mapOverlay FRIDGE_SOH_VALUE). ------ And then check with Kostas to see if we can just throw an exception if any data builder tries to request data from dhis getAnalytics with >1 programCode, >1 DE.  

`programCodes` -> `programCode`.
`if (programCodes typeof Array) throw 'no'`